packages/big-design/src/components/Tooltip/Tooltip.tsx

```diff
@@ -13,90 +13,78 @@ export interface TooltipProps extends React.HTMLAttributes<HTMLDivElement> {
   inline?: boolean;
 }
 
-interface State {
-  visible: boolean;
-}
-
-export class Tooltip extends React.PureComponent<TooltipProps, State> {
-  static defaultProps: Partial<TooltipProps> = {
-    placement: 'top',
-    inline: true,
-  };
-
-  state = {
-    visible: false,
-  };
-
-  private tooltipContainer?: HTMLDivElement;
+export const Tooltip: React.FC<TooltipProps> = memo(({ children, inline = true, modifiers, trigger, ...props }) => {
+  const [isVisible, setIsVisible] = useState(false);
+  const [tooltipContainer, setTooltipContainer] = useState<HTMLDivElement | null>(null);
 
-  componentDidMount() {
-    this.tooltipContainer = document.createElement('div');
-    document.body.appendChild(this.tooltipContainer);
-  }
+  useEffect(() => {
+    const container = document.createElement('div');
 
-  componentWillUnmount() {
-    if (this.tooltipContainer) {
-      document.body.removeChild(this.tooltipContainer);
-    }
-  }
-
-  render() {
-    const { children, trigger, inline } = this.props;
+    document.body.appendChild(container);
+    setTooltipContainer(container);
+  }, []);
 
-    return (
-      <Manager>
-        <Reference>
-          {({ ref }) => (
-            <StyledTooltipTrigger
-              inline={inline}
-              onBlur={this.hideTooltip}
-              onFocus={this.showTooltip}
-              onKeyDown={this.onKeyDown}
-              onMouseEnter={this.showTooltip}
-              onMouseLeave={this.hideTooltip}
-              ref={ref}
-            >
-              {trigger}
-            </StyledTooltipTrigger>
-          )}
-        </Reference>
-        {this.tooltipContainer
-          ? createPortal(
-              <Popper
-                placement={this.props.placement}
-                modifiers={{ offset: { offset: '0, 8' }, ...this.props.modifiers }}
-                eventsEnabled={this.state.visible}
-              >
-                {({ placement, ref, style }) =>
-                  this.state.visible && (
-                    <StyledTooltip ref={ref} style={style} data-placement={placement}>
-                      {this.renderContent(children)}
-                    </StyledTooltip>
-                  )
-                }
-              </Popper>,
-              this.tooltipContainer,
-            )
-          : null}
-      </Manager>
-    );
-  }
+  useEffect(() => {
+    return () => {
+      if (tooltipContainer) {
+        document.body.removeChild(tooltipContainer);
+      }
+    };
+  }, [tooltipContainer]);
 
-  private renderContent = (content: React.ReactNode) => {
+  const renderContent = (content: React.ReactNode) => {
```

Comment:
🍹 Since you have access to `children` here I would use it directly instead of passing it.

Reply:
Yeah, I went back and forth on whether to pass it or use it directly. This implementation was only to cover the edge case that this function could be used elsewhere in future refactors (though unlikely). Will change it back. 👍